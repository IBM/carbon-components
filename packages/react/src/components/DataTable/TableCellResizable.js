/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import PropTypes from 'prop-types';
import React from 'react';
import { useColumnResizing } from './tools/columnResize';

const TableCellResizable = ({ children, colKey, isResizable, ...rest }) => {
  const { colWidth } = useColumnResizing(colKey);

  if (isResizable && colWidth && colKey) {
    return (
      <td style={{ width: colWidth + 'px' }} {...rest}>
        {children}
      </td>
    );
  }

  return <td {...rest}>{children}</td>;
};

TableCellResizable.propTypes = {
  /**
   * Pass in children that will be embedded in the table header label
   */
  children: PropTypes.node,

  /**
   * key for the column as defined in the header data
   */
  colKey: PropTypes.string,

  /**
   * `false` If true, will apply resizable styles
   */
  isResizable: PropTypes.bool,
};

TableCellResizable.displayName = 'TableCellResizable';

export default TableCellResizable;
