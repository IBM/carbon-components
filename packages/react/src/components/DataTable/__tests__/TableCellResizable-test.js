/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Table, TableBody, TableRow } from '..';
import TableCellResizable from '../TableCellResizable';

jest.mock('../tools/columnResize', () => {
  const hookReturnVal = {
    colWidth: 100,
    ref: null,
    columnKeyResizeActive: null,
    initColumnResizing: jest.fn(),
    cleanupColumnResizing: jest.fn(),
    startResizeAction: jest.fn(),
    endResizeAction: jest.fn(),
    resizeColumn: jest.fn(),
  };
  return {
    hookReturnVal,
    useResizedColumnWidth: () => hookReturnVal,
  };
});

describe('DataTable.TableCellResizable', () => {
  it('should render', () => {
    const wrapper = mount(
      <Table>
        <TableBody>
          <TableRow>
            <TableCellResizable />
          </TableRow>
        </TableBody>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with resizable columns', () => {
    const wrapper = mount(
      <Table>
        <TableBody>
          <TableRow>
            <TableCellResizable key="aKey" colKey="aKey" />
          </TableRow>
        </TableBody>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
