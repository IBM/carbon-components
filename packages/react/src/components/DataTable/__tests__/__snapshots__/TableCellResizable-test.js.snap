// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableCellResizable should render 1`] = `
<Table
  isSortable={false}
>
  <table
    className="bx--data-table bx--data-table--no-border"
  >
    <TableBody
      aria-live="polite"
    >
      <tbody
        aria-live="polite"
      >
        <TableRow>
          <tr>
            <TableCellResizable>
              <td />
            </TableCellResizable>
          </tr>
        </TableRow>
      </tbody>
    </TableBody>
  </table>
</Table>
`;

exports[`DataTable.TableCellResizable should render with resizable columns 1`] = `
<Table
  isSortable={false}
>
  <table
    className="bx--data-table bx--data-table--no-border"
  >
    <TableBody
      aria-live="polite"
    >
      <tbody
        aria-live="polite"
      >
        <TableRow>
          <tr>
            <TableCellResizable
              colKey="aKey"
              key="aKey"
            >
              <td />
            </TableCellResizable>
          </tr>
        </TableRow>
      </tbody>
    </TableBody>
  </table>
</Table>
`;
